### PR TITLE
add websocket handshake to server

### PR DESCRIPTION
**Pull request by rbren**
This just creates an "agent" using a websocket echo server. The basic flow is this:
* Client connects to server
* Client sends `start` command
* Server starts up an agent inside a docker container
* Server connects to agent websocket
* Server forwards all messages between client and agent
* Client disconnects
* Server closes connections and stops the docker container

### CODE REVIEW

File: server/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,4 @@
 fastapi
 uvicorn
+uvicorn[standard]
```
Comment (neubig) [inline server/requirements.txt:3]: Just to confirm, do we need both of these?

**Review by neubig — APPROVED** [review]
This looks great! I approve, but just wanted to confirm if we needed both imports of uvicorn.

File: server/requirements.txt (hunk 1 above)
Comment (yimothysu) [reply server/requirements.txt:3]: `uvicorn[standard]` is a superset of `uvicorn` that includes WebSocket support, so `uvicorn` can be removed from line 2.

**Comment by huybery** [comment]
@rbren hey, I just sent the slack invite like your email and look forward to you joining to talk to us. :)

File: server/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
 fastapi
 uvicorn
```
Comment (rbren) [inline server/requirements.txt:2]:
```suggestion
```

File: server/requirements.txt (hunk 1 above)
Comment (rbren) [reply server/requirements.txt]: 👍 thanks! Done

**Pull request merged**